### PR TITLE
FEATURE: BCO-1300 Add model for analytical dashboard and visualization object

### PR DESCRIPTION
**Pull request by hung-nguyen-hoang**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/694/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/695/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/696/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/698/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/699/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/701/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/702/ : SUCCESS in 6s

File: lib/gooddata/models/metadata/analytical_visualization_object.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      end
+    end
+
+    # Method used for replacing values in their state according to mapping.
+    # Can be used to replace any values but it is typically used to replace
+    # the URIs. Returns a new object of the same type.
+    #
+    # @param [Array<Array>]mapping specifying what should be exchanged for what. As mapping should be used output of GoodData::Helpers.prepare_mapping.
+    # @return [GoodData::AnalyticalVisualizationObject]
+    def replace(mapping)
```
Comment (danh-ung) [inline lib/gooddata/models/metadata/analytical_visualization_object.rb:36]: We should have unit test for this function.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/706/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/707/ : SUCCESS in 18s

File: lib/gooddata/models/metadata/analytical_visualization_object.rb (hunk 1 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/models/metadata/analytical_visualization_object.rb]: Checked the function **replace(mapping)** https://github.com/gooddata/gooddata-ruby/blob/develop/lib/gooddata/models/metadata.rb#L199 in parent class already cover same as this function. So delete this method and no need any unit tests for this function.

File: lib/gooddata/models/metadata/analytical_dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+#
+# Copyright (c) 2010-2021 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
```
Comment (sangtm) [inline lib/gooddata/models/metadata/analytical_dashboard.rb:5]: Please update all license BSD-style to BSD-3-CLAUSE

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1759

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/531/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/2/ : SUCCESS in 1m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/87/ : SUCCESS in 1m 21s (non-voting)

**Pull request merged**